purpleair-aqi.js

```diff
@@ -89,184 +128,191 @@ const levelAttributes = [
-    gradient = new LinearGradient();
-    console.log('light mode');
-}
+    const aqiTrend = getAQITrend(stats);
+    console.log(aqiTrend);
+
+    const epaPM = computePM(data);
+    console.log(`EPA PM is ${epaPM}`);
+
+    const aqi = aqiFromPM(epaPM);
+    const level = calculateLevel(aqi);
+    const aqiText = aqi.toString();
+    console.log(`AQI is ${aqi}`);
+
+    const isDarkMode = Device.isUsingDarkAppearance();
+
+    const startColor = new Color(
+      isDarkMode ? level.darkStartColor : level.startColor
+    );
+    const endColor = new Color(
+      isDarkMode ? level.darkEndColor : level.endColor
+    );
+    const textColor = new Color(
+      isDarkMode ? level.darkTextColor : level.textColor
+    );
+    const gradient = new LinearGradient();
+
+    console.log(`${isDarkMode ? "dark" : "light"} mode`);
 
     gradient.colors = [startColor, endColor];
     gradient.locations = [0.0, 1];
     console.log(gradient);
 
-    wg.backgroundGradient = gradient;
+    listWidget.backgroundGradient = gradient;
 
-    let header = wg.addText("AQI" + theTrend);
+    const header = listWidget.addText(`AQI${aqiTrend}`);
     header.textColor = textColor;
     header.font = Font.regularSystemFont(15);
 
-    let content = wg.addText(aqiText);
+    const content = listWidget.addText(aqiText);
     content.textColor = textColor;
     content.font = Font.semiboldRoundedSystemFont(30);
 
-    let wordLevel = wg.addText(level.label);
+    const wordLevel = listWidget.addText(level.label);
     wordLevel.textColor = textColor;
     wordLevel.font = Font.boldSystemFont(25);
-    wordLevel.minimumScaleFactor = 0.3
+    wordLevel.minimumScaleFactor = 0.3;
 
-    wg.addSpacer(10);
+    listWidget.addSpacer(10);
 
-    let location = wg.addText(data.loc);
+    const location = listWidget.addText(data.loc);
     location.textColor = textColor;
     location.font = Font.mediumSystemFont(12);
 
-    let updatedAt = new Date(data.ts * 1000)
-      .toLocaleTimeString([], { hour: '2-digit', minute: '2-digit'});
-    let ts = wg.addText(`Updated ${updatedAt}`);
-    ts.textColor = textColor;
-    ts.font = Font.lightSystemFont(10);
-
-    wg.addSpacer(10);
-
-    let purpleMap = 'https://www.purpleair.com/map?opt=1/i/mAQI/a10/cC0&select=' + SENSOR_ID + '#14/' + data.lat + '/' + data.lon;
-    wg.url = purpleMap;
-  } catch (e) {
-    console.log(e);
-
-    let err = wg.addText(`${e}`);
-    err.textColor = Color.red();
-    err.textOpacity = 30;
-    err.font = Font.regularSystemFont(10);
+    const updatedAt = new Date(data.ts * 1000).toLocaleTimeString([], {
+      hour: "2-digit",
+      minute: "2-digit",
+    });
+    const widgetText = listWidget.addText(`Updated ${updatedAt}`);
+    widgetText.textColor = textColor;
+    widgetText.font = Font.lightSystemFont(10);
+
+    listWidget.addSpacer(10);
+
+    const purpleMapUrl = `https://www.purpleair.com/map?opt=1/i/mAQI/a10/cC0&select=${SENSOR_ID}#14/${data.lat}/${data.lon}`;
```

Comment:
I find this type of templating to be more readable than `'string' + variable + 'otherString'`.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals